File: app/src/main/java/com/example/unima_l003/demoyoutubeplayer/adapters/SearchVideoAdapter.java
package com.example.unima_l003.demoyoutubeplayer.adapters;

import android.view.ViewGroup;

import com.example.unima_l003.demoyoutubeplayer.abstracts.NMViewHolder;
import com.example.unima_l003.demoyoutubeplayer.objects.Video;

import java.util.List;


/**
 * unima
 * Created by michael on 7/4/2016.
 */
public class SearchVideoAdapter extends VideoAdapter {

    public SearchVideoAdapter(List<Video> items) {
        super(items);
    }

    @Override
    public NMViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return super.onCreateViewHolder(parent, viewType);
    }

    @Override
    public void onBindViewHolder(NMViewHolder holder, int position) {
        if (holder instanceof VideoVH) {
            int positionVideo = getItemVideoPosition(position);
            super.onBindViewHolder(holder, positionVideo);
        }
    }

    @Override
    public int getItemViewType(int position) {
        return super.getItemViewType(position);
    }





}

File: app/src/main/java/com/example/unima_l003/demoyoutubeplayer/adapters/SearchVideoAdapter.kt
package com.example.unima_l003.demoyoutubeplayer.adapters

import android.view.ViewGroup

import com.example.unima_l003.demoyoutubeplayer.abstracts.NMViewHolder
import com.example.unima_l003.demoyoutubeplayer.objects.Video


/**
 * unima
 * Created by michael on 7/4/2016.
 */
class SearchVideoAdapter(items: List<Video>) : VideoAdapter(items) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NMViewHolder =
            super.onCreateViewHolder(parent, viewType)

    override fun onBindViewHolder(holder: NMViewHolder, position: Int) {
        if (holder is VideoAdapter.VideoVH) {
            val positionVideo = getItemVideoPosition(position)
            super.onBindViewHolder(holder, positionVideo)
        }
    }

    override fun getItemViewType(position: Int): Int = super.getItemViewType(position)


}
